File: tests/test_api.js
#!/usr/bin/env node
/**
 * API Testing Script for AI导航
 * Tests all deployed endpoints and reports their status
 */

const BASE_URL = "http://localhost:8787";  // 本地测试，生产环境需修改

async function testHealthCheck() {
  console.log("=".repeat(60));
  console.log("Testing Health Check (GET /)");
  console.log("=".repeat(60));
  try {
    const response = await fetch(`${BASE_URL}/`);
    const text = await response.text();
    console.log(`✓ Status Code: ${response.status}`);
    console.log(`✓ Response: ${text}`);
    return response.status === 200;
  } catch (e) {
    console.log(`✗ Error: ${e.message}`);
    return false;
  }
}

async function testRecommend() {
  console.log("\n" + "=".repeat(60));
  console.log("Testing Smart Recommend (POST /api/recommend)");
  console.log("=".repeat(60));
  try {
    const payload = { query: "我想剪辑一个视频" };
    console.log(`Request: ${JSON.stringify(payload)}`);
    const response = await fetch(`${BASE_URL}/api/recommend`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(payload)
    });
    const result = await response.json();
    console.log(`✓ Status Code: ${response.status}`);
    console.log(`✓ Response: ${JSON.stringify(result, null, 2)}`);
    return response.status === 200 && result.recommendations;
  } catch (e) {
    console.log(`✗ Error: ${e.message}`);
    return false;
  }
}

async function testCases() {
  console.log("\n" + "=".repeat(60));
  console.log("Testing Cases List (GET /api/cases)");
  console.log("=".repeat(60));
  try {
    const response = await fetch(`${BASE_URL}/api/cases`);
    const result = await response.json();
    console.log(`✓ Status Code: ${response.status}`);
    console.log(`✓ Response: ${JSON.stringify(result, null, 2)}`);
    console.log(`✓ Total cases: ${result.cases?.length || 0}`);
    return response.status === 200 && result.cases;
  } catch (e) {
    console.log(`✗ Error: ${e.message}`);
    return false;
  }
}

async function main() {
  console.log("\n🚀 Starting API Tests");
  console.log(`Base URL: ${BASE_URL}\n`);

  const results = {
    "Health Check": await testHealthCheck(),
    "Smart Recommend": await testRecommend(),
    "Cases List": await testCases()
  };

  console.log("\n" + "=".repeat(60));
  console.log("Test Summary");
  console.log("=".repeat(60));
  for (const [testName, passed] of Object.entries(results)) {
    const status = passed ? "✓ PASSED" : "✗ FAILED";
    console.log(`${testName}: ${status}`);
  }

  const total = Object.keys(results).length;
  const passed = Object.values(results).filter(Boolean).length;
  console.log(`\nTotal: ${passed}/${total} tests passed`);

  process.exit(passed === total ? 0 : 1);
}

main();

File: tests/test_api.py
#!/usr/bin/env python3
"""
API Testing Script for AI导航
Tests all deployed endpoints and reports their status
"""

import requests
import json
import sys

BASE_URL = "http://localhost:8787"  # 本地测试，生产环境需修改

def test_health_check():
    """Test GET / endpoint"""
    print("=" * 60)
    print("Testing Health Check (GET /)")
    print("=" * 60)
    try:
        response = requests.get(f"{BASE_URL}/", timeout=10)
        print(f"✓ Status Code: {response.status_code}")
        print(f"✓ Response: {response.text}")
        return response.status_code == 200
    except Exception as e:
        print(f"✗ Error: {e}")
        return False

def test_recommend():
    """Test POST /api/recommend endpoint"""
    print("\n" + "=" * 60)
    print("Testing Smart Recommend (POST /api/recommend)")
    print("=" * 60)
    try:
        payload = {"query": "我想剪辑一个视频"}
        print(f"Request: {json.dumps(payload, ensure_ascii=False)}")
        response = requests.post(
            f"{BASE_URL}/api/recommend",
            json=payload,
            headers={"Content-Type": "application/json"},
            timeout=10
        )
        print(f"✓ Status Code: {response.status_code}")
        result = response.json()
        print(f"✓ Response: {json.dumps(result, indent=2, ensure_ascii=False)}")
        return response.status_code == 200 and "recommendations" in result
    except Exception as e:
        print(f"✗ Error: {e}")
        return False

def test_cases():
    """Test GET /api/cases endpoint"""
    print("\n" + "=" * 60)
    print("Testing Cases List (GET /api/cases)")
    print("=" * 60)
    try:
        response = requests.get(f"{BASE_URL}/api/cases", timeout=10)
        print(f"✓ Status Code: {response.status_code}")
        result = response.json()
        print(f"✓ Response: {json.dumps(result, indent=2, ensure_ascii=False)}")
        print(f"✓ Total cases: {len(result.get('cases', []))}")
        return response.status_code == 200 and "cases" in result
    except Exception as e:
        print(f"✗ Error: {e}")
        return False

def main():
    print("\n🚀 Starting API Tests")
    print(f"Base URL: {BASE_URL}\n")

    results = {
        "Health Check": test_health_check(),
        "Smart Recommend": test_recommend(),
        "Cases List": test_cases()
    }

    print("\n" + "=" * 60)
    print("Test Summary")
    print("=" * 60)
    for test_name, passed in results.items():
        status = "✓ PASSED" if passed else "✗ FAILED"
        print(f"{test_name}: {status}")

    total = len(results)
    passed = sum(results.values())
    print(f"\nTotal: {passed}/{total} tests passed")

    sys.exit(0 if passed == total else 1)

if __name__ == "__main__":
    main()
